source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,179 @@
+==============
+Retrieve Data
+==============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to retrieve data from your MongoDB
+database. To retrieve data, use read operations.
+
+Read operations allow you to do the following:
+
+- :ref:`Retrieve a subset of documents from your collection <retrieve-find>`
+- :ref:`Perform transformations on retrieved documents from your collection <retrieve-aggregate>`
+- :ref:`Monitor real-time changes to your database <retrieve-watch>`
+
+Paint Orders
+~~~~~~~~~~~~~
+
+The examples in the following sections are about a paint store that
+keeps track of orders they need to fulfill. The paint store keeps track
+of orders with the ``color`` and ``qty`` fields in their
+``paint_orders`` collection: 
+
+.. code-block:: json
+
+    { "_id": 1, "color": "purple", "qty": 10 }
+    { "_id": 2, "color": "green", "qty": 8 }
+    { "_id": 3, "color": "purple", "qty": 4 }
+    { "_id": 4, "color": "green", "qty": 11 }
+
+Read operations
+---------------
+
+.. _retrieve-find:
+
+Find
+~~~~
+
+Use the find operation to retrieve a subset of your existing data in
+MongoDB. To further specify which documents to retrieve, you can sort,
+skip, and limit the documents or fields.
+
+To perform a find operation, use the ``find()`` method on a
+collection. This method searches a collection for documents that
+match the query filter you provide. For more information on how to
+specify a query, see our :doc:`Specify a Query
+</fundamentals/crud/query-document>` guide. 
+
+To access the results, you can iterate through the results using a
+cursor. For more information on query filters, see our :doc:`Access Data
```

Comment:
Does your example use a cursor to iterate through the results? If it's important enough to mention, should the example also demonstrate a cursor?

Reply:
I kept this here because we talk about the way for each is used on the cursor page (acts like a cursor even though it isn't a cursor object)

Reply:
Usually when you mention that your code example accomplishes something "using <something>", the reader expects to see it in the code or a description of how it's used if not explicitly in the code. While the driver probably communicates with a cursor provided by the server while you iterate through the result object when you call forEach on the AggregateIterable, it's a couple levels removed from a cursor.

I think it's ok to describe accessing the results through a cursor and not show it in your example, but it would be helpful to have the reader understand that's the case. For an example of how you can use a MongoCursor idiomatically, see:
https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCursor.html

You can modify your existing code (if you want to show MongoCursor) by calling `iterator()` on your Iterable and then iterating over that as shown in the MongoCursor javadoc example.

If you feel strongly about not providing this info or incorporating it into your example, that's fine, not a blocker.